File: Vanilla/sources/zdoc/media/lua/shared/Library/StorySound.lua
---@class StorySound : zombie.radio.StorySounds.StorySound
---@field protected name String
---@field protected baseVolume float
StorySound = {}

---@public
---@return float
function StorySound:getBaseVolume() end

---@public
---@return long
---@overload fun(arg0:float)
---@overload fun(arg0:float, arg1:float, arg2:float, arg3:float, arg4:float)
---@overload fun(arg0:float, arg1:float, arg2:float, arg3:float, arg4:float, arg5:float)
function StorySound:playSound() end

---@public
---@param arg0 float
---@return long
function StorySound:playSound(arg0) end

---@public
---@param arg0 float
---@param arg1 float
---@param arg2 float
---@param arg3 float
---@param arg4 float
---@return long
function StorySound:playSound(arg0, arg1, arg2, arg3, arg4) end

---@public
---@param arg0 float
---@param arg1 float
---@param arg2 float
---@param arg3 float
---@param arg4 float
---@param arg5 float
---@return long
function StorySound:playSound(arg0, arg1, arg2, arg3, arg4, arg5) end

---@public
---@return String
function StorySound:getName() end

---@public
---@param arg0 float
---@return void
function StorySound:setBaseVolume(arg0) end

---@public
---@return StorySound
function StorySound:getClone() end

---@public
---@param arg0 String
---@return void
function StorySound:setName(arg0) end

File: Vanilla/sources/zomboid/zombie/radio/StorySounds/StorySound.java
package zombie.radio.StorySounds;

import zombie.characters.IsoPlayer;
import zombie.iso.Vector2;

public final class StorySound {
   protected String name = null;
   protected float baseVolume = 1.0F;

   public StorySound(String var1, float var2) {
      this.name = var1;
      this.baseVolume = var2;
   }

   public long playSound() {
      Vector2 var1 = SLSoundManager.getInstance().getRandomBorderPosition();
      return SLSoundManager.Emitter.playSound(this.name, this.baseVolume, var1.x, var1.y, 0.0F, 100.0F, SLSoundManager.getInstance().getRandomBorderRange());
   }

   public long playSound(float var1) {
      return SLSoundManager.Emitter.playSound(this.name, var1, IsoPlayer.getInstance().x, IsoPlayer.getInstance().y, IsoPlayer.getInstance().z, 10.0F, 50.0F);
   }

   public long playSound(float var1, float var2, float var3, float var4, float var5) {
      return this.playSound(this.baseVolume, var1, var2, var3, var4, var5);
   }

   public long playSound(float var1, float var2, float var3, float var4, float var5, float var6) {
      return SLSoundManager.Emitter.playSound(this.name, this.baseVolume * var1, var2, var3, var4, var5, var6);
   }

   public String getName() {
      return this.name;
   }

   public void setName(String var1) {
      this.name = var1;
   }

   public float getBaseVolume() {
      return this.baseVolume;
   }

   public void setBaseVolume(float var1) {
      this.baseVolume = var1;
   }

   public StorySound getClone() {
      return new StorySound(this.name, this.baseVolume);
   }
}
